src/ToricVarieties/LineBundles/methods.jl

```diff
@@ -0,0 +1,81 @@
+############################
+# Line bundle cohomologies
+############################
+
+@doc Markdown.doc"""
+    all_cohomologies(l::ToricLineBundle)
+
+Computes the dimension of all sheaf cohomologies of the toric line bundle `l`.
+
+# Examples
+```jldoctest
+julia> dP3 = del_pezzo(3)
+A normal, non-affine, smooth, projective, gorenstein, q-gorenstein, fano, 2-dimensional toric variety without torusfactor
+
+julia> all_cohomologies(ToricLineBundle(dP3, [1,2,3,4]))
+3-element Vector{fmpz}:
+ 11
+ 0
+ 0
+```
+"""
+function all_cohomologies(l::ToricLineBundle)
+    # check if we can apply cohomCalg
+    v = variety(l)
+    if !((issmooth(v) && iscomplete(v)) || (issimplicial(v) && isprojective(v)))
+        throw(ArgumentError("cohomCalg only applies to toric varieties that are either smooth, complete or simplicial, projective."))
+    end
+    
+    # extract vector of currently-known cohomology dimensions (or create it if necessary)
+    return get_attribute!(l, :all_cohomologies) do
+        
+        # obtain the command string
+        class = vec([fmpz(x) for x in divisor_class(l).coeff])
+        command = command_string(v, class)
+        
+        # execute cohomCalg
+        out = Pipe()
+        err = Pipe()
+        process = run(pipeline(ignorestatus(`$(cohomCalg_jll.cohomcalg()) --integrated --in=$(command)`), stdout=out, stderr=err))
+        close(out.in)
+        close(err.in)
+        
+        # was there an error?
+        stderr = String(read(err))
+        code = process.exitcode
+        if code != 0
+            throw(ErrorException("cohomCalg encountered the error " * stderr))
+        end
+        
+        # read out the result
+        stdout = String(read(out))
+        result = [fmpz(parse(Int,c)) for c in split(chop(chop(split(stdout, "{" )[4])), ",")]
```

Comment:
I would suggest to insert an (possible abbreviated/simplified) example of what data you are expecting here and trying to parse. This will make it easier to debug problems here in the future, e.g. caused by a newer `cohomCalg` version or whatever else may change. It can also help anybody who wants to tweak this code (e.g. to simplify it, make it more robust, extend it for supporting additional features...). That "someone" of course may well be you, just in a few months/years... ;-)
```suggestion
        stdout = String(read(out))
        result = [fmpz(parse(Int,c)) for c in split(chop(chop(split(stdout, "{" )[4])), ",")]
```

Reply:
Thank you. I have just added an example which explains the strategy. This is indeed a good idea. :+1: